File: simplify-path/simplify-path.cs
public class Solution {
    public string SimplifyPath(string path) {
        
        Stack<string> stack=new Stack<string>();
        
        string[] pathArray=path.Split("/");
        
        for(int i=0;i<pathArray.Length;i++)
        {
           // console.WriteLine(pathArray[i]);
            if(pathArray[i]==".." && stack.Count > 0)
            {
                stack.Pop();
            }
            else if(pathArray[i]!=".." && pathArray[i]!="" && pathArray[i]!=".")
            {
                stack.Push(pathArray[i]);
            }
        }
        
        string ans="";
        if(stack.Count>0)
        {
            while(stack.Count>0)
            {
                ans="/"+stack.Pop()+ans;
            }
        }
        else
            ans="/";
        
        return ans;
        
    }
}

File: simplify-path/simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] folders=path.split(String.valueOf('/'));
        
        Stack<String> stack=new Stack<>();
        
        for(int i=0;i<folders.length;i++)
        {
           // System.out.println(folders[i]);
            if(folders[i].equals("."))
                continue;
            if(folders[i].equals("..") && stack.size()>0)
            {
                stack.pop();
            }
            else if(!folders[i].isEmpty() && !folders[i].equals(".."))
            {
                stack.push(folders[i]);
            }
        }
        
        String resultPath="";
        
        while(stack.size()>0)
        {
            resultPath="/"+stack.pop()+resultPath;
        }
        
        return resultPath.isEmpty() ? "/" : resultPath;
    }
}